src/applications/personalization/preferences/actions/index.js

```diff
@@ -179,6 +181,10 @@ export function updatePreferences(
   saveFunc = savePreferences,
   deleteFunc = deletePreferences,
 ) {
+  recordEvent({
+    event: 'dashboard-preferences-saved-successful',
+    ...transformPreferencesForAnalytics(benefitsData),
+  });
```

Comment:
It's entirely possible recording the analytics event will need to be moved from the action creator to the component level. This is because this action is fired from different components and the analytics folks might want to record different analytics event for different user actions. ie, there are multiple ways for a user to update their benefit selections and we may need to record them all as unique events. TBD.

Reply:
I don't have any good ideas for this. @jbalboni might

- we could use special action creators just for analytics events and push them in as props (this would separate analytics from the behavior of the application but it widens the interface of the component)
- we could curry the analytics events into existing props in `mapDispatchToProps` 
- we could use metadata in actions that a middleware could pick up e.g. [redux-analytics](https://github.com/markdalgleish/redux-analytics)

I guess we should start validating our analytics with a GA debugger and see if there's any benefit to having a pattern. I assume analytics stuff is going to be pretty volatile this year and having GA calls all over components is going to be harder to maintain.

Reply:
Yeah, I don't have a great solution for this, either. A redux middleware would be the cleanest, but that assumes that all user actions have a Redux equivalent and aren't just affecting local component state. That's mostly how we want to write our apps, I think, but we don't always do it.

Reply:
Lots of things we are tracking are just clicks on `button` or `a` or `Link` elements. So we might need to maintain at least two different ways to record events: the simple clicks and then the deeper data-changing/redux-type changes.